Source/Scene/Scene.js

```diff
@@ -1632,6 +1640,46 @@ define([
         }
     }
 
+    function executeShadowMapCommands(scene) {
+        if (!scene.shadowMap.enabled) {
+            return;
+        }
+
+        var context = scene.context;
+        var uniformState = context.uniformState;
+        var shadowMap = scene.shadowMap;
+        var renderState = shadowMap.renderState;
+        var passState = shadowMap.passState;
+        var cameraScene = scene.camera;
+
+        // Clear shadow depth
+        shadowMap.clearCommand.execute(context);
+
+        // Render from the shadow's camera
+        uniformState.updateCamera(shadowMap.camera);
+
+        var shadowMapCommands = scene._frustumCommandsList[0]; // TODO : temporary solution
+        if (defined(shadowMapCommands)) {
+            // TODO : Execute only opaque and translucent commands for now.
+            var startPass = Pass.OPAQUE;
+            var endPass = Pass.TRANSLUCENT;
+            for (var pass = startPass; pass < endPass; ++pass) {
+                var commands = shadowMapCommands.commands[pass];
+                var length = shadowMapCommands.indices[pass];
+                for (var i = 0; i < length; ++i) {
+                    var command = commands[i];
+                    // TODO : Use the shader as-is for now since color mask is false. Later will need a specific shadow cast shader.
+                    //var shaderProgram = shadowMap.createShadowCastProgram(command.shaderProgram);
+                    //executeCommand(command, scene, context, passState, renderState, shaderProgram);
+                    executeCommand(command, scene, context, passState, renderState);
+                }
+            }
+        }
+
+        // Switch back to the scene's camera
+        uniformState.updateCamera(cameraScene);
```

Comment:
As I mentioned in another comment, instead of this set/restore paradigm, I would rather the start of each pass just have to set this.